http2curl.go

```diff
@@ -27,11 +27,25 @@ func bashEscape(str string) string {
 
 // GetCurlCommand returns a CurlCommand corresponding to an http.Request
 func GetCurlCommand(req *http.Request) (*CurlCommand, error) {
+	if req.URL == nil {
+		return nil, fmt.Errorf("getCurlCommand: invalid request, req.URL is nil")
+	}
+
 	command := CurlCommand{}
 
 	command.append("curl")
 
-	if req.URL != nil && req.URL.Scheme == "https" {
+	schema := req.URL.Scheme
+	requestURL := req.URL.String()
+	if schema == "" {
+		schema = "http"
+		if req.TLS != nil {
+			schema = "https"
+		}
+		requestURL = schema + "://" + req.Host + req.URL.Path
```

Comment:
if only + req.URL.Path will drop query. I think if we just add default schema when schema=="", I think
`requestURL = schema + "://" + requestURL` is ok

Reply:
Oh I see the title "the issue req.URL.Host is empty ", so we should process schema is empty and URL.host is empty separately. I'll fire another PR